### PR TITLE
Initial Components pane for React

### PR DESCRIPTION
**Pull request by calebstdenis**
Related issue: #3792 

### Summary of Changes
* Add a "Components" pane that appears when pausing in a React component. For now, it just lists the component and all its ancestors all the way up to the root.
* This is hidden behind a feature flag for now.

Todo in a future PR: 
* Ability to "select" components. Right now it's just a plain list.
* Improve styling?
* Improve handling for container components created with `connect`? 

### Test Plan
1. Hit a breakpoint at [Frame/index.js#174](https://github.com/calebstdenis/debugger.html/blob/12363ee0bddab3819b781e98bacf00b530015937/src/components/SecondaryPanes/Frames/index.js#L174)
2. Observe the "Components" pane as in the screenshot below

### Screenshots/Videos
image

### CODE REVIEW

File: src/actions/pause/extra.js [hunk 1]
```diff
@@ -38,21 +52,7 @@ async function getImmutableProps(expression: string, evaluate) {
...
-
-      if (originalClass) {
-        props.react = { displayName: originalClass.name };
-      }
-    }
-
-    if (!props.react) {
-      props.react = await getReactProps(evaluate);
-    }
+    props.react = await getReactProps(evaluate);
```
Comment (calebstdenis) [inline src/actions/pause/extra.js]: @jasonLaster @mmcote Is this OK? It was the easiest way to get what I wanted but I can put it back if needed.

File: src/components/SecondaryPanes/index.js [hunk 2]
```diff
@@ -175,6 +176,17 @@ class SecondaryPanes extends Component<Props, State> {
     };
   }
 
+  getComponentStackItem() {
+    return {
+      header: "Components",
```
[the remark this answers is not in the document]
Comment (calebstdenis) [reply src/components/SecondaryPanes/index.js]: fixed in 156ba1ea34e787b1854f39c087e88ae96de17526

File: src/utils/prefs.js [hunk 3]
```diff
@@ -50,6 +51,7 @@ if (isDevelopment()) {
   pref("devtools.debugger.features.outline", true);
   pref("devtools.debugger.features.column-breakpoints", true);
   pref("devtools.debugger.features.replay", true);
+  pref("devtools.debugger.features.component-stack", true);
```
[the remark this answers is not in the document]
Comment (calebstdenis) [reply src/utils/prefs.js]: fixed in ca230f1fd39774518baf1bd84e55b9e03f88ec8b

File: src/actions/pause/extra.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (calebstdenis) [reply src/actions/pause/extra.js]: fixed in ca230f1

File: assets/panel/debugger.properties [hunk 4]
```diff
@@ -392,6 +392,9 @@ callStack.collapse=Collapse rows
 # message to show more of the frames.
 callStack.expand=Expand rows
 
+# LOCALIZATION NOTE (components.header): Components right sidebar pane header.
```
Comment (flodolo) [inline assets/panel/debugger.properties]: Maybe something a bit more descriptive? e.g. 'Header pane in the right sidebar for React Components', assuming it's only for React

Comment (calebstdenis) [reply assets/panel/debugger.properties]: How's this? f2c5442fcb3a64517a412e11947a7544166cd03f
'Header for the React Components pane in the right sidebar.'

For now, it's only for React.

File: src/actions/pause/extra.js [hunk 5]
```diff
@@ -7,19 +7,35 @@
...
   );
-
-  return {
-    displayName: reactDisplayName.result
-  };
+  const items =
+    componentNames.result.preview && componentNames.result.preview.items;
+  if (items) {
+    return {
+      displayName: items[0],
```
Comment (jasonLaster) [inline src/actions/pause/extra.js]: hmm, i'd prefer not overriding displayName. The rationale is if you are debugging a site that is minified, but has source maps looking up the class name from the symbols table is not minified. Crazy, right?!?!

File: src/actions/pause/extra.js [hunk 6]
```diff
@@ -50,9 +66,7 @@ async function getExtraProps(getState, expression, result, evaluate) {
       }
     }
 
-    if (!props.react) {
-      props.react = await getReactProps(evaluate);
-    }
+    merge(props.react, await getReactProps(evaluate));
```
Comment (jasonLaster) [inline src/actions/pause/extra.js]: can we use splat here `{...props.react, ...await getReactProps(evaluate) }`

File: src/actions/pause/extra.js (hunk 5 above)
Comment (calebstdenis) [reply src/actions/pause/extra.js]: Makes sense! Fixed in 106eeaa17ebc51250d1927ad053372968c2cbacc

File: src/actions/pause/extra.js (hunk 6 above)
Comment (calebstdenis) [reply src/actions/pause/extra.js]: sure can. 106eeaa17ebc51250d1927ad053372968c2cbacc

File: src/actions/pause/extra.js (hunk 5 above)
Comment (calebstdenis) [reply src/actions/pause/extra.js]: actually wait 106eeaa still changes the behavior... bc60352db7980803b7cded66a2063a30e3a99155 should do the job.

File: src/actions/pause/extra.js (hunk 6 above)
Comment (calebstdenis) [reply src/actions/pause/extra.js]: update: bc60352db7980803b7cded66a2063a30e3a99155

**Comment by jasonLaster** [comment]
@calebstdenis can you rebase and fix the the circle tests

File: assets/panel/debugger.properties [hunk 7]
```diff
@@ -392,6 +392,10 @@ callStack.collapse=Collapse rows
 # message to show more of the frames.
 callStack.expand=Expand rows
 
+# LOCALIZATION NOTE (components.header): Header for the React Components pane
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: this should be `Header for the Framwork Components`

we want to add Ember soon :)

**Comment by calebstdenis** [comment]
Had to fix some Jest tests 487edd6a2297b529f2ae3d573db880f71d1d4168

**Pull request merged**